pkg/systemd/host.js

```diff
@@ -334,43 +335,66 @@ PageServer.prototype = {
         var pmcd_service = service.proxy("pmcd");
         var pmlogger_service = service.proxy("pmlogger");
         var pmlogger_promise;
-
-        $("#server-pmlogger-switch").on("change", function(ev) {
-            var val = $(this).onoff('value');
-            if (pmlogger_service.exists) {
-                if (val) {
-                    pmlogger_promise = cockpit.all(pmcd_service.enable(),
-                           pmcd_service.start(),
-                           pmlogger_service.enable(),
-                           pmlogger_service.start()).
-                        fail(function (error) {
-                            console.warn("Enabling pmlogger failed", error);
-                        });
-                } else {
-                    pmlogger_promise = cockpit.all(pmlogger_service.disable(),
-                           pmlogger_service.stop()).
-                        fail(function (error) {
-                            console.warn("Disabling pmlogger failed", error);
-                        });
-                }
-                pmlogger_promise.always(function() {
-                    pmlogger_promise = null;
-                    refresh_pmlogger_state();
-                });
-            }
-        });
+        var pmlogger_exists = false;
 
         function refresh_pmlogger_state() {
-            if (!pmlogger_service.exists)
-                $('#server-pmlogger-onoff-row').hide();
-            else if (!pmlogger_promise) {
-                $("#server-pmlogger-switch").onoff('value', pmlogger_service.enabled);
-                $('#server-pmlogger-onoff-row').show();
-            }
+            /* HACK: The pcp packages on Ubuntu and Debian inclue SysV init
+             * scripts in /etc, which stay around when removing (as opposed to
+             * purging) the package. This makes systemd think that there's
+             * still a unit there, and pmlogger_service.exists will be true.
```

Comment:
Clarification: This is not a "thinko" of systemd -- there *is* a valid and enabled SysV init script, and thus also a unit. There is no way/API to tell about a generic init.d script whether it is "backed" by a package or not.

Reply:
Makes sense. Rephrased.